tests/__sf_section_dir.test.fish

```diff
@@ -0,0 +1,191 @@
+		set_color normal
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_dir)
+end
+
+test "Correctly truncates a home subdirectory"
+	(
+		cd ~/.tmp-spacefish/dir1/
+
+		set_color --bold fff
+		echo -n "in "
+		set_color normal
+		set_color --bold cyan
+		echo -n "~/.tmp-spacefish/dir1"
+		set_color normal
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_dir)
+end
+
+test "Correctly truncates a deeply nested home subdirectory"
+	(
+		cd ~/.tmp-spacefish/dir1/dir2
+
+		set_color --bold fff
+		echo -n "in "
+		set_color normal
+		set_color --bold cyan
+		echo -n ".tmp-spacefish/dir1/dir2"
+		set_color normal
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_dir)
+end
+
+#
+# Root directory
+#
+
+test "Correctly truncates root directory"
+	(
+		cd /
+
+		set_color --bold fff
+		echo -n "in "
+		set_color normal
+		set_color --bold cyan
+		echo -n "/"
+		set_color normal
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_dir)
+end
+
+test "Correctly truncates a root subdirectory"
+	(
+		cd /usr
+
+		set_color --bold fff
+		echo -n "in "
+		set_color normal
+		set_color --bold cyan
+		echo -n "/usr"
+		set_color normal
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_dir)
+end
+
+test "Correctly truncates a deeply nested root subdirectory"
+	(
+		cd /tmp/tmp-spacefish/dir1/dir2
+
+		set_color --bold fff
+		echo -n "in "
+		set_color normal
+		set_color --bold cyan
+		echo -n "tmp-spacefish/dir1/dir2"
+		set_color normal
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_dir)
+end
+
+#
+# Git directory
+#
+
+test "Correctly truncates the root of a git directory"
+	(
+		cd /tmp/tmp-spacefish
+		command git init >/dev/null
+
+		set_color --bold fff
+		echo -n "in "
+		set_color normal
+		set_color --bold cyan
+		echo -n "tmp-spacefish"
+		set_color normal
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_dir)
+end
+
+test "Correctly truncates a git subdirectory"
+	(
+		cd /tmp/tmp-spacefish
+		command git init >/dev/null
+		cd /tmp/tmp-spacefish/dir1
+
+		set_color --bold fff
+		echo -n "in "
+		set_color normal
+		set_color --bold cyan
+		echo -n "tmp-spacefish/dir1"
+		set_color normal
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_dir)
+end
+
+test "Correctly truncates a deeply nested git subdirectory"
+	(
+		cd /tmp/tmp-spacefish
+		command git init >/dev/null
+		cd /tmp/tmp-spacefish/dir1/dir2/dir3
+
+		set_color --bold fff
+		echo -n "in "
+		set_color normal
+		set_color --bold cyan
+		echo -n "dir1/dir2/dir3"
+		set_color normal
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_dir)
+end
+
+test "Correctly truncates the root of a git directory within another"
+	(
+		cd /tmp/tmp-spacefish
+		command git init >/dev/null
+
+		cd /tmp/tmp-spacefish/dir1
+		command git init >/dev/null
```

Comment:
This gets repetitive, maybe a function could optimize this. Something like:

```fish
function run_no_error_ouput cmd
  command cmd >/dev/null
end

run_no_error_output (git init)
```

Obviously pseudo code, not sure it could actually work like this

Reply:
I don't personally have any problems with how it is. Wrapping it in a function would end up with just as much repetition, just repetition of a function call.

Reply:
I made it assuming it would be used with different commands, you can also just have it call command `git init >/dev/null` all the time instead for a shorter function call. Lil bit overkill though I agree.